lib/checkautovariables.cpp

```diff
@@ -58,7 +58,7 @@ static bool isPtrArg(const Token *tok)
 static bool isArrayArg(const Token *tok, const Settings& settings)
 {
     const Variable *var = tok->variable();
-    return (var && var->isArgument() && var->isArray() && !settings.library.isentrypoint(var->scope()->className));
+    return (var && var->isArgument() && var->isArray() && (!var->scope() || !settings.library.isentrypoint(var->scope()->className)));
```

Comment:
I wonder if the scope is wrong in this case, i.e. should `f` be set in the test case?